File: backend/src/main/java/com/board/backend/service/BoardCommentService.java
package com.board.backend.service;

import com.board.backend.model.BoardCommentDto;

public interface BoardCommentService {
    BoardCommentDto.ResponseList list(long bno, BoardCommentDto.RequestList form);
    BoardCommentDto.Response get(long no);
    BoardCommentDto.Response insert(long bno, BoardCommentDto.Create form);
    BoardCommentDto.Response update(long no, BoardCommentDto.Update form);
    BoardCommentDto.Response delete(long no);
    void deleteAll(long bno);
}

File: frontend/src/common/services/BoardCommentService.js
import { AppConstants } from '../index'
import HttpClient from './HttpClient'

const SERVICE_URL = `${AppConstants.BASE_URL}/board`

const BoardCommentService = {
  list: async (bno, form) => {
    const url = `${SERVICE_URL}/${bno}/comments`
    return await HttpClient.get(url, form)
  },
  get: async (bno, form) => {
    const url = `${SERVICE_URL}/${bno}/comment/${form.no}`
    return await HttpClient.get(url)
  },
  create: async (bno, form) => {
    const url = `${SERVICE_URL}/${bno}/comment`
    return await HttpClient.post(url, form)
  },
  update: async (bno, form) => {
    const url = `${SERVICE_URL}/${bno}/comment`
    return await HttpClient.put(url, form)
  },
  delete: async (bno, no) => {
    const url = `${SERVICE_URL}/${bno}/comment/${no}`
    return await HttpClient.delete(url)
  },
}

export default BoardCommentService
